### PR TITLE
Set region if configured in AWS config object.

### PR DESCRIPTION
**Pull request by chsing-br**

### CODE REVIEW

File: tests/AmazonConnection.test.js [hunk 1]
```diff
@@ -74,5 +74,32 @@ describe('AmazonConnection', function () {
       assert.strictEqual(req.headers['X-Amz-Date'], '20190816T205118Z')
       assert.strictEqual(req.headers.Authorization, 'AWS4-HMAC-SHA256 Credential=foo/20190816/us-east-1/es/aws4_request, SignedHeaders=content-length;content-type;date;host;x-amz-date;x-amz-security-token, Signature=5daf2e5d70fe61002e12ad3d4d8dcfcda64bb477ce764af1c30967393d83ba1f')
     })
+
+    it('sets the region when region is configured in the AWS config object', function () {
+      const awsConfig = new AWS.Config({
```
Comment (Nick-Rivera-Meredith) [inline tests/AmazonConnection.test.js:79]: The name `awsConfig` will shadow the globally declared `awsConfig` object.

File: tests/AmazonConnection.test.js [hunk 2]
```diff
@@ -74,5 +74,32 @@ describe('AmazonConnection', function () {
...
+        accessKeyId: 'foo',
+        region: 'us-west-2',
+        secretAccessKey: 'bar',
+        sessionToken: 'baz'
+      })
+
+      const AmazonConnection = require('../src/AmazonConnection')(awsConfig)
```
Comment (Nick-Rivera-Meredith) [inline tests/AmazonConnection.test.js:86]: This will shadow the global `AmazonConnection` object.

**Review by Nick-Rivera-Meredith — CHANGES_REQUESTED** [review]
🥥

File: tests/AmazonConnection.test.js (hunk 1 above)
Comment (chsing-br) [reply tests/AmazonConnection.test.js]: Updated.

File: tests/AmazonConnection.test.js (hunk 2 above)
Comment (chsing-br) [reply tests/AmazonConnection.test.js]: Updated.

**Review by Nick-Rivera-Meredith — APPROVED** [review]

**Pull request merged**